src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -711,6 +705,226 @@ public String[] executeSimple(Object[][] params) {
 						"\t\tset {_money} to formatNumber({money::%sender's uuid%})",
 						"\t\tsend \"Your balance: %{_money}%\" to sender")
 			.since("2.10");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("mean", new Parameter[]{
+			new Parameter<>("numbers", DefaultClasses.NUMBER, false, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				double total = 0;
+				for (Object object : params[0]) {
+					Number number = (Number) object;
+					if (number instanceof Double doubleNumber && (doubleNumber.isInfinite() || doubleNumber.isNaN())) {
+						Skript.error("You cannot get the mean of a set of number that includes 'infinity' or 'nan'.");
+						return null;
```

Comment:
You can't skript.error here, this is runtime, not parsetime.

Reply:
maybe it should just skip infinity and nans